spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -145,46 +154,35 @@ public SpannerConverter getSpannerConverter() {
 	}
 
 	@Override
-	public <T> List<T> queryAll(Class<T> entityClass, Sort sort) {
-		return queryAll(entityClass, sort, null);
-	}
-
-	@Override
-	public <T> List<T> queryAll(Class<T> entityClass, Sort sort,
-			SpannerQueryOptions options) {
-		Assert.notNull(sort, "sort must not be null!");
-
-		StringBuilder stringBuilder = new StringBuilder();
+	public <T> List<T> queryAll(Class<T> entityClass, SpannerQueryOptions options) {
 		SpannerPersistentEntity<?> persistentEntity = this.mappingContext
 				.getPersistentEntity(entityClass);
-		stringBuilder.append("SELECT * FROM " + persistentEntity.tableName() + " ");
-		SpannerStatementQueryExecutor.buildOrderBy(persistentEntity, stringBuilder, sort);
-		if (options != null) {
+		String sql = "SELECT * FROM " + persistentEntity.tableName();
+		return query(entityClass, applySortingPagingQueryOptions(options, sql), null,
+				null, options);
+	}
+
+	public String applySortingPagingQueryOptions(SpannerQueryOptions options,
+			String sql) {
+		String r = applySort(options.getSort(), sql);
 			if (options.hasLimit()) {
-				stringBuilder.append(" LIMIT " + options.getLimit());
+			r += " LIMIT " + options.getLimit();
 			}
 			if (options.hasLimit()) {
-				stringBuilder.append(" OFFSET " + options.getOffset());
+			r += " OFFSET " + options.getOffset();
 			}
-		}
-		stringBuilder.append(";");
-		return query(entityClass, Statement.of(stringBuilder.toString()), options);
+		return r;
 	}
 
-	@Override
-	public <T> Page<T> queryAll(Class<T> entityClass, Pageable pageable,
-			SpannerQueryOptions options) {
-		Assert.notNull(pageable, "Pageable must not be null!");
-
-		Long count = count(entityClass);
-		List<T> list = queryAll(entityClass, pageable.getSort(), options);
-		return new PageImpl(list, pageable, count);
-	}
-
-	@Override
-	public <T> Page<T> queryAll(Class<T> entityClass, Pageable pageable) {
-		return queryAll(entityClass, pageable, new SpannerQueryOptions()
-				.setOffset(pageable.getOffset()).setLimit(pageable.getPageSize()));
+	private String applySort(Sort sort, String sql) {
+		if (sort == null || sort.isUnsorted()) {
+			return sql;
+		}
+		String s = "SELECT * FROM (" + sql + ") ORDER BY ";
```

Comment:
Cool idea for guarding against existing clauses :) I hope there are no extra cost to it (it seems that way, that Spanner is smart enough to optimize issues from little experiments on my console)